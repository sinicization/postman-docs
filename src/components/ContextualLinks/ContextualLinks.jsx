import React from 'react';
import DynamicLink from '../Shared/DynamicLink';
import recentBlogPosts from '../../../bff-data/blog.json';

import './ContextualLinks.scss';

const ContextualLinks = ({ links }) => (
  <div className="contextual-links">
    {
      links.map((item) => {
        if (item.type === 'link') {
          return (
            <div className="contextual-links__link" key={item.name}>
              <DynamicLink url={item.url} name={item.name} />
            </div>
          );
        }
        if (item.type === 'subtitle') {
          return (
            <div className="contextual-links__subtitle" key={item.name}>
              {item.name}
            </div>
          );
        }
        if (item.type === 'dynamic_blog' && item.blog_tag) {
          if (Object.keys(recentBlogPosts).length === 0) {
            // If recentBlogPosts.length === 0, then either there is no .env.development,
            // it has a bad blog url, or the endpoint returned something bad
            return (
              <div className="contextual-links__alert" role="alert">
                <p>
                  您目前处于开发模式.
                  动态博客文章将不会在本地显示.
                  <a style={{ fontSize: 'inherit' }} href="https://github.com/postmanlabs/postman-docs/blob/develop/CONTRIBUTING.md#to-use-the-dynamic-blog-posts-feature" target="_blank" rel="noopener noreferrer">有关详细信息，请参见贡献文档</a>
                  .
                </p>
              </div>
            );
          }
          if (recentBlogPosts.index && recentBlogPosts.index[item.blog_tag]) {
            const dynamicLinks = recentBlogPosts.index[item.blog_tag].slice(0, 3).map((post) => (
              <div className="contextual-links__link" key={post.title}>
                <DynamicLink url={post.link.url} name={post.title} />
              </div>
            ));
            return (
              <>
                <div className="contextual-links__subtitle" key={item.name}>
                  {item.name}
                </div>
                {dynamicLinks}
              </>
            );
          }

          // If there are no related blog posts within blogs.json
          // (written in /bff-data/ at build time), we simply return nothing.

          // This will likely be the case when you are developing locally,
          // and do not have a .env.development file with the BFF endpoints.
          return null;
        }

        return (
          <div className="contextual-links__section" key={Math.random()}>{item.name}</div>
        );
      })
    }
  </div>
);

export default ContextualLinks;
