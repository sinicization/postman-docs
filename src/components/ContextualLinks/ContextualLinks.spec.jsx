import React from 'react';
import renderer from 'react-test-renderer';
import ContextualLinks from './ContextualLinks';

describe('ContextualLinks', () => {
  it('renders correctly', () => {
    const testLinks = [
      {
        type: 'section',
        name: '先决条件',
      },
      {
        type: 'link',
        name: '下载并安装',
        url: 'https://www.postman.com/downloads/',
      },
      {
        type: 'subtitle',
        name: '相关博客文章',

      },
      {
        type: 'link',
        name: 'home',
        url: '/',
      },
    ];

    const tree = renderer
      .create(<ContextualLinks links={testLinks} />)
      .toJSON();
    expect(tree.children[0].props.className).toBe('contextual-links__section');
    expect(tree.children[1].props.className).toBe('contextual-links__link');
    expect(tree).toMatchSnapshot();
  });
});
