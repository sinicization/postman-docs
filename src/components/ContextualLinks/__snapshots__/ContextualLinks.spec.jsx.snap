// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualLinks renders correctly 1`] = `
<div
  className="contextual-links"
>
  <div
    className="contextual-links__section"
  >
    先决条件
  </div>
  <div
    className="contextual-links__link"
  >
    <a
      className="dynamic-link__external"
      href="https://www.postman.com/downloads/"
    >
      下载并安装
    </a>
  </div>
  <div
    className="contextual-links__subtitle"
  >
    相关博客文章
  </div>
  <div
    className="contextual-links__link"
  >
    <a
      className="dynamic-link__internal"
      href="/"
    >
      home
    </a>
  </div>
</div>
`;
